### PR TITLE
Adding icon picker and display, scroll bar missing border still and i…

### PR DESCRIPTION
**Pull request by Oglopf**
…con not updating when selected still

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201735133575781/1202271754222051) by [Unito](https://www.unito.io)

### CODE REVIEW

File: apps/dashboard/app/models/project.rb [hunk 1]
```diff
@@ -17,10 +18,13 @@ def all
...
 
     def find(project_path)
+      # previously the @proj_name kicked off a whole chain of events firing
+      # now, we are just getting that manifest entry back only
+      #
       full_path = dataroot.join(project_path)
       return nil unless full_path.directory?
 
-      Project.new({ name: full_path.basename })
+      Project.new( name: full_path.basename )
```
Comment (Oglopf) [inline apps/dashboard/app/models/project.rb:27]: This currently breaks at the update as there is only a `name` to be served back to populate fields.

File: apps/dashboard/app/models/project.rb [hunk 2]
```diff
@@ -3,6 +3,7 @@
 class Project
   include ActiveModel::Model
   include ActiveModel::Validations
+  include ActiveModel::Validations::Callbacks
```
Comment (Oglopf) [inline apps/dashboard/app/models/project.rb:6]: This was from a previous strategy that was backed out and now this needs removed.

**Review by gbsoftwaresolutions — CHANGES_REQUESTED** [review]
The only thing that I found was user experience issue.

If there is an "Error" during initial create/edit of project, the name and description revert to previous value (empty on create, latest saved values on edit)

File: apps/dashboard/app/models/project.rb [hunk 3]
```diff
@@ -30,34 +27,52 @@ def dataroot
...
   end
 
-  validates :directory, format: {
+  validates :directory, presence: true
+  validates :name, format: {
     with:    /\A[\w-]+\z/,
     message: I18n.t('dashboard.jobs_project_name_validation')
   }
 
+  validates :icon, format: {
```
[the remark this answers is not in the document]
Comment (Oglopf) [reply apps/dashboard/app/models/project.rb:44]: Yeah, good catch!

**Comment by Oglopf** [comment]
> The only thing that I found was user experience issue.
> 
> If there is an "Error" during initial create/edit of project, the name and description revert to previous value (empty on create, latest saved values on edit)

Good catch. Got the parameters passing back now, but I need to clean up a bit and write a few tests to catch this too.

File: apps/dashboard/app/controllers/projects_controller.rb [hunk 4]
```diff
@@ -13,7 +13,7 @@ def index
 
   # GET /projects/new
   def new
-    @project = Project.new
+    params[:name].nil? || params[:name].empty? ? @project = Project.new : @project = Project.new({ name: params[:name], icon: params[:icon] })
```
Comment (Oglopf) [inline apps/dashboard/app/controllers/projects_controller.rb:16]: Still adjusting this to be more robust.

**Review by gerald-byrket — APPROVED** [review]

**Pull request merged**